indexing-service/src/main/java/io/druid/indexing/common/TaskLock.java

```diff
@@ -31,19 +31,24 @@
   private final String dataSource;
   private final Interval interval;
   private final String version;
+  private final Integer priority;
+  private boolean exclusiveLock;
```

Comment:
should this be volatile ?
I would prefer making it part of constructor arg ? 
also I think serde should preserve the status whether the lock was exclusive or not.